### DISCUSSION TITLE
Update Entry Date by Code

### DISCUSSION THREAD

**Discussion by sewid** [statamic/cms#6186] [Q&A] [answered]
Hi there,

I am looking for a way to randomize the Entry dates from a collection (a "blog" collection containing posts").

What I tried so far:

```php
foreach (Entry::whereCollection('blog')->all() as $entry) {
    $entry->date = Carbon::today()->subDays(rand(0, 365));
    $entry->save();
}
```
and
```php
foreach (Entry::whereCollection('blog')->all() as $entry) {
    $entry->set('date', Carbon::today()->subDays(rand(0, 365)));
    $entry->save();
}
```

But it seems, I cannot update the post date. Only the field 'date' is being updated, but not the filename containing the date. Can somebody please help me how to randomize the post dates?

Thanks!
Sebastian

**Comment by duncanmcclean** [comment, marked as answer]
You can do `$entry->date(Carbon::today()->subDays(rand(0, 365))))`

**Reply by sewid** [reply]
Great, this is working - thanks a lot!

**Reply by duncanmcclean** [reply]
No worries!
